lib/config/secret.js

```diff
@@ -0,0 +1,28 @@
+module.exports = getSecret
+
+var internals = getSecret.internals = {}
+internals.randomstring = require('randomstring')
+
+function getSecret (state, callback) {
+  state.db.config.get()
+
+  .then(function (config) {
+    if (config.secret) {
+      state.secret = config.secret
+      return callback()
+    }
+
+    config.secret = internals.randomstring.generate({
+      charset: 'hex'
+    })
+
+    state.db.config.set(config)
+
+    .then(function () {
```

Comment:
As below, to me this `.then()` within a `.then()` looks a bit funky. Personally I'd prefer `throw`ing cases that should skip later steps, or by returning a rejected `Promise`, and then handling those cases in the `.catch()`.